src/aten/sycl/DistributionTemplates.h

```diff
@@ -0,0 +1,680 @@
+#pragma once
+
+#include <ATen/Dispatch.h>
+#include <ATen/Dispatch_v2.h>
+#include <ATen/ExpandBase.h>
+#include <ATen/OpMathType.h>
+#include <aten/sycl/Loops.h>
+#include <aten/sycl/MemoryAccess.h>
+#include <aten/sycl/OffsetCalculator.h>
+#include <aten/sycl/Philox4x32.h>
+#include <comm/DeviceProperties.h>
+#include <comm/Runtime.h>
+
+namespace at {
+namespace native {
+namespace xpu {
+
+using namespace at::xpu;
+
+const uint32_t rand4_engine_calls = 4;
+
+struct PhiloxState {
+  PhiloxState() = default;
+  // Called if graph capture is not underway
+  PhiloxState(uint64_t seed, uint64_t offset) {
+    seed_ = seed;
+    offset_.val = offset;
+  }
+  // Called if graph capture is underway
+  PhiloxState(
+      uint64_t seed,
+      int64_t* offset_extragraph,
+      uint32_t offset_intragraph) {
+    seed_ = seed;
+    offset_.ptr = offset_extragraph;
+    offset_intragraph_ = offset_intragraph;
+    captured_ = true;
+  }
+
+  union Payload {
+    uint64_t val;
+    int64_t* ptr;
+  };
+
+  uint64_t seed_ = 0;
+  Payload offset_;
+  uint32_t offset_intragraph_ = 0;
+  bool captured_ = false;
+};
+
+inline std::tuple<uint64_t, uint64_t> philox_unpack(PhiloxState arg) {
+  if (arg.captured_) {
+    // static_cast avoids "warning: invalid narrowing conversion from "long" to
+    // "unsigned long".
+    // *(arg.offset_.ptr) is a broadcast load of a single int64_t to the entire
+    // kernel. For most threads' reads it will hit in cache, so it shouldn't
+    // hurt performance.
+    return std::make_tuple(
+        arg.seed_,
+        static_cast<uint64_t>(*(arg.offset_.ptr) + arg.offset_intragraph_));
+  } else {
+    return std::make_tuple(arg.seed_, arg.offset_.val);
+  }
+}
+
+inline std::tuple<uint64_t, uint32_t, uint32_t> calc_execution_policy(
+    int64_t total_elements) {
+  auto group_size = syclGpuHWThreadsPerEU() * syclMaxSubGroupSize();
+  auto num_groups = (total_elements + group_size - 1) / group_size;
+  auto hw_max_groups = syclMaxWorkItemsPerTile() / group_size;
+  num_groups = num_groups > hw_max_groups ? hw_max_groups : num_groups;
+  // number of times random will be generated per thread, to offset philox
+  // counter in thc random state
+  uint64_t counter_offset =
+      ((total_elements - 1) / (group_size * num_groups * rand4_engine_calls) +
+       1) *
+      rand4_engine_calls;
+  return std::make_tuple(counter_offset, num_groups, group_size);
+}
+
+// Just follow loops.h design
+template <
+    typename accscalar_t,
+    int unroll_factor,
+    typename dist_t,
+    typename transform_t,
+    typename item_t>
+inline void distribution_elementwise_kernel(
```

Comment:
Just inline the code into `DistributionElementwiseKernel` and remove `distribution_elementwise_kernel`.